tools/clang/tools/dxc/dxc.cpp

```diff
@@ -62,6 +62,43 @@
 #include <algorithm>
 #include <unordered_map>
 
+// SPIRV Change Starts
+#ifdef ENABLE_SPIRV_CODEGEN
+#include "spirv-tools/libspirv.hpp"
+
+static bool DisassembleSpirv(const CComPtr<IDxcBlob> &binaryBlob,
+                             CComPtr<IDxcLibrary> library,
+                             IDxcBlobEncoding **assemblyBlob) {
+  if (!binaryBlob)
+    return true;
+
+  size_t num32BitWords = (binaryBlob->GetBufferSize() + 3) / 4;
+  std::string binaryStr((char *)binaryBlob->GetBufferPointer(),
+                        binaryBlob->GetBufferSize());
+  binaryStr.resize(num32BitWords * 4, 0);
+
+  std::vector<uint32_t> words;
+  words.resize(num32BitWords, 0);
+  memcpy(words.data(), binaryStr.data(), binaryStr.size());
+
+  std::string assembly;
+  spvtools::SpirvTools spirvTools(SPV_ENV_VULKAN_1_0);
+  spirvTools.SetMessageConsumer(
+    [](spv_message_level_t, const char *, const spv_position_t &,
+      const char *message) { fprintf(stderr, "%s\n", message); });
```

Comment:
fprintf might not be what you want if the user has a console with a different encoding. Just something to think about - you may or may not care about this.

Reply:
Removed. We don't really care error messages here (unlikely to happen anyway).